Draft-Accepted/RFCXXXX-Command-Shell-Integration.md

```diff
@@ -0,0 +1,287 @@
+---
+RFC: RFCNNNN
+Author: Steve Lee
+Status: Draft
+SupercededBy: NA
+Version: 1.0
+Area: Interactive Shell
+Comments Due: June 1, 2023
+Plan to implement: Yes
+---
+
+# Command Shell Integration
+
+Executables can provide a more integrated experience with a particular shell by indicating features it supports which the shell can leverage.
+
+Shell integration feature areas:
+
+- Structured output and formatting
+- Tab completion
+- Help
+- Predictors (PowerShell)
+- Feedback Providers (PowerShell)
+
+Traditionally, POSIX shells have pre-defined filesystem locations where tools would put their tab completion scripts or help content.
+However, this typically works for the system and not for a particular user who doesn't have administrative rights to the system.
```

Comment:
> Fish automatically searches through any directories in the list variable `$fish_complete_path`, and any completions defined are automatically loaded when needed.   - From [fish completion doc](https://fishshell.com/docs/current/completions.html#where-to-put-completions)

It's very much like the `PSModulePath` concept we have in PowerShell, and it works fine for a user who doesn't have administrator privileges.

Reply:
I'm sure you're very aware of the problems with `PSModulePath` (partly due to it being modified by PowerShell and not just used as-is), but in this case, the manifest JSON would just be in `$env:PATH` and is shell-agnostic.

Reply:
Adding additional text on looking for manifests in symlinks or reparse point target locations